webapi/src/main/scala/org/knora/webapi/routing/HealthRouteWithZIOHttp.scala

```diff
@@ -0,0 +1,117 @@
+        case AppState.UpdatingRepository     => unhealthy("Updating repository. Please retry later.")
+        case AppState.RepositoryUpToDate     => unhealthy("Repository up to date. Please retry later.")
+        case AppState.CreatingCaches         => unhealthy("Creating caches. Please retry later.")
+        case AppState.CachesReady            => unhealthy("Caches ready. Please retry later.")
+        case AppState.UpdatingSearchIndex    => unhealthy("Updating search index. Please retry later.")
+        case AppState.SearchIndexReady       => unhealthy("Search index ready. Please retry later.")
+        case AppState.LoadingOntologies      => unhealthy("Loading ontologies. Please retry later.")
+        case AppState.OntologiesReady        => unhealthy("Ontologies ready. Please retry later.")
+        case AppState.WaitingForIIIFService  => unhealthy("Waiting for IIIF service. Please retry later.")
+        case AppState.IIIFServiceReady       => unhealthy("IIIF service ready. Please retry later.")
+        case AppState.WaitingForCacheService => unhealthy("Waiting for cache service. Please retry later.")
+        case AppState.CacheServiceReady      => unhealthy("Cache service ready. Please retry later.")
+        case AppState.MaintenanceMode        => unhealthy("Application is in maintenance mode. Please retry later.")
+        case AppState.Running                => healthy
+      }
+    )
+
+  private def createResponse(result: HealthCheckResult): UIO[Response] =
+    ZIO.succeed(
+      Response
+        .json(
+          JsObject(
+            "name"     -> JsString("AppState"),
+            "severity" -> JsString("non fatal"),
+            "status"   -> JsString(status(result.status)),
+            "message"  -> JsString(result.message)
+          ).toString()
+        )
+        .setStatus(statusCode(result.status))
+    )
+
+  private def status(s: Boolean): String = if (s) "healthy" else "unhealthy"
+
+  private def statusCode(s: Boolean): Status = if (s) Status.Ok else Status.ServiceUnavailable
+
+  private case class HealthCheckResult(name: String, severity: String, status: Boolean, message: String)
+
+  private def unhealthy(message: String) =
+    HealthCheckResult(
+      name = "AppState",
+      severity = "non fatal",
+      status = false,
+      message = message
+    )
+
+  private val healthy =
+    HealthCheckResult(
+      name = "AppState",
+      severity = "non fatal",
+      status = true,
+      message = "Application is healthy"
+    )
+}
+
+/**
+ * Provides the '/healthZ' endpoint serving the health status.
+ */
+final case class HealthRouteWithZIOHttp(state: State) extends HealthCheckWithZIOHttp {
+
+  /**
+   * Returns the route.
+   */
+  val route: Http[State, Nothing, Request, Response] =
+    Http.collectZIO[Request] { case Method.GET -> !! / "healthZ" =>
+      for {
+        //  ec    <- ZIO.executor.map(_.asExecutionContext) // TODO leave this for reference about how to get the execution context
```

Comment:
this is not really a TODO, I would say:

```suggestion
        //  ec    <- ZIO.executor.map(_.asExecutionContext) // leave this for reference about how to get the execution context
```